target/linux/qualcommax/ipq50xx/base-files/lib/upgrade/platform.sh

```diff
@@ -69,8 +71,100 @@ platform_check_image() {
 	return 0;
 }
 
+glinet_validate_firmware() {
+	local type=$(identify_magic_long $(get_magic_long $1))
```

Comment:
(Note: a lot of my comments will be syntax fixes pointed by [ShellCheck](https://www.shellcheck.net/): I'm not sure if this is used in the OpenWrt repository, but this file at least seemed to be compliant. Feel free to ignore them if you don't want to use it)

```suggestion
	local type
        type=$(identify_magic_long "$(get_magic_long "$1")")
```

https://www.shellcheck.net/wiki/SC2155